verification/abi.go

```diff
@@ -126,3 +154,370 @@ type GetTaggedTCIResp[Digest DigestAlgorithm] struct {
 	CumulativeTCI Digest
 	CurrentTCI    Digest
 }
+
+// DPEABI is a connection to a DPE instance, parameterized by hash algorithm and ECC curve.
+type DPEABI[CurveParameter Curve, Digest DigestAlgorithm] struct {
+	transport    Transport
+	Profile      Profile
+	MajorVersion uint16
+	MinorVersion uint16
+	VendorId     uint32
+	VendorSku    uint32
+	MaxTciNodes  uint32
+	Flags        uint32
+}
+
+// DPEABI256 is a client that implements DPE_PROFILE_IROT_P256_SHA256
+type DPEABI256 = DPEABI[NISTP256Parameter, SHA256Digest]
+
+// DPEABI384 is a client that implements DPE_PROFILE_IROT_P384_SHA384
+type DPEABI384 = DPEABI[NISTP384Parameter, SHA384Digest]
+
+// dpeProfileImplementsTypeConstraints checks that the requested DPEABI type constraints are compatible with the DPE profile.
+func dpeProfileImplementsTypeConstraints[C Curve, D DigestAlgorithm](profile Profile) error {
+	// Test that the expected value types produced by each DPE profile can be assigned to variables of type C and D
+	var c C
+	var d D
+	switch profile {
+	case ProfileP256SHA256:
```

Comment:
nit/highly optional: just thinking about the error messages here. If you derive the expected profile based on C and D, you can compare that derived profile with the actual one, returning either:

* "Client requested (Curve C, Digest D), which is not a valid DPE profile"
* "Client requested DPE Profile X, but DPE implements Y"

Reply:
Done.